app/controllers/uploads_controller.rb

```diff
@@ -4,14 +4,14 @@ def new
   end
 
   def create
-    @upload = Upload.new(upload_params)
+    @folder = Folder.find(params[:upload][:folder_id])
+    @upload = @folder.uploads.new(upload_params)
```

Comment:
I made this change because paperclip wasn't working, but it didn't fix it. I kept it this way because the Upload model belongs to the Folder model and I've gotten errors on previous projects when I haven't gone through the hierarchy.